pkg/resource/container_image.go

```diff
@@ -0,0 +1,426 @@
+				indexManifest.Manifests[index].Platform.OS, indexManifest.Manifests[index].Platform.Architecture)
+		}
+		imgRef := imageSource.Reference().DockerReference().Name() + "@" + indexManifest.Manifests[index].Digest.String()
+
+		return r.cloneRemoteResource(path, imgRef)
+	}
+
+	img, err := imgRef.NewImage(ctx, sysCtx)
+	if err != nil {
+		log.Fatalf("could not load image to retrieve labels: %v", err)
+	}
+	defer img.Close()
+
+	config, err := img.OCIConfig(ctx)
+	if err != nil {
+		log.Fatalf("could not get image config to retrieve labels: %v", err)
+	}
+	r.labels = config.Config.Labels
+
+	configJSON, err := json.MarshalIndent(config, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to create string from configjson: %v", err)
+	}
+	err = r.writeFile("config.json", configJSON)
+	if err != nil {
+		return fmt.Errorf("failed to write config to clonepath: %v", err)
+	}
+
+	imgManifest, err := manifest.FromBlob(rawManifest, manifestType)
+	if err != nil {
+		return fmt.Errorf("could not parse manifest: %v", err)
+	}
+
+	cache := blobinfocache.DefaultCache(sysCtx)
+	for _, layer := range imgManifest.LayerInfos() {
+		skipLayer := false
+		for _, exclude := range r.options.Exclusions {
+			if exclude == layer.Digest.String() {
+				logger.Debug("skipping layer %s", layer.Digest.String())
+				skipLayer = true
+			}
+		}
+		if skipLayer {
+			continue
+		}
+		logger.Debug("downloading layer %s", layer.Digest.String())
+
+		layerDir := filepath.Join(path, layer.Digest.Hex())
+		err = os.MkdirAll(layerDir, 0700)
+		if err != nil {
+			return fmt.Errorf("could not create layer directory: %v", err)
+		}
+
+		blobInfo := types.BlobInfo{
+			Digest: layer.Digest,
+			Size:   layer.Size,
+		}
+		layerBlob, _, err := imageSource.GetBlob(ctx, blobInfo, cache)
+		if err != nil {
+			return fmt.Errorf("could not download layer blob: %v", err)
+		}
+
+		err = r.decompress(layerBlob, layer.Digest.Hex(), layer.MediaType, layer.Size)
+		if err != nil {
+			return fmt.Errorf("could not decompress layer: %v", err)
+		}
+		err = layerBlob.Close()
+		if err != nil {
+			return fmt.Errorf("could not close layer: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *ContainerImage) writeFile(filename string, content []byte) error {
+	file, err := os.OpenFile(filepath.Join(r.clonePath, filename), os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600) // #nosec G304
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	_, err = file.Write(content)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// The decompression process is a little more involved so separated out.
+func (r *ContainerImage) decompress(t io.Reader, layer string, mediaType string, size int64) error {
+	// The maximum file size should be less than 10x the layer size.
+	size = size * 10
+	path := filepath.Join(r.ClonePath(), layer)
+
+	var tarReader *tar.Reader
+
+	if strings.Contains(strings.ToLower(mediaType), "gzip") {
```

Comment:
Generally this is using a docker schema. Regardless, it would still end up as a string comparison. 
It could be a switch, it could use consts from the library. Though, this could potentially make it more fragile. All the specs share the behavior that if it is a gzip'd it has gzip in it, if it does not it is just a tar. This could change if zstd ever becomes common place. It is a rarity for now and everything still ships gzip first in manifests. 

I've added support if we come across zstd and changed from `Contains` to `HasSuffix`, as it is quicker.